### PR TITLE
WDPD-19 Create basket items

### PR DESCRIPTION
**Pull request by ps-cgrach**
### This PR:

* add basket handling with discounts
* also adds wrapping, gift cards, etc. to the basket

### How to test

* get sure to enable everything a basket can contain
  * Packaging
  * gift card
  * shipping costs
  * payment costs

go through the whole purchase flow with Paypal and see the request includes all those fields.

### Jira tickets

https://jira.parkside.at/browse/WDPD-19
https://jira.parkside.at/browse/WDPD-99

### CODE REVIEW

File: Core/BasketHelper.php [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+namespace Wirecard\Oxid\Core;
+
+use OxidEsales\Eshop\Application\Model\Article;
```
Comment (ps-holler) [inline Core/BasketHelper.php]: This doesn't seem to be used.

File: Extend/Payment_Gateway.php [hunk 2]
```diff
@@ -13,13 +13,18 @@
 use \OxidEsales\Eshop\Application\Model\Article;
 use \OxidEsales\Eshop\Application\Model\Country;
 use \OxidEsales\Eshop\Application\Model\Basket;
+use \OxidEsales\Eshop\Application\Model\BasketItem;
```
Comment (ps-holler) [inline Extend/Payment_Gateway.php]: This doesn't seem to be used.

File: translations/de/wirecardpaymentgateway_de_lang.php [hunk 3]
```diff
@@ -460,7 +462,9 @@
     "vault_use_existing_text" => "Gespeicherte Kreditkarte verwenden",
     "vault_use_new_text" => "Neue Kreditkarte verwenden",
     "vault_valid_until_text" => "Gültig bis",
+    "voucher" => "Gutschein",
     "Wirecard AG" => "Wirecard AG",
     "Wirecard WooCommerce Extension" => "Wirecard WooCommerce Extension",
+    "wrapping" => "Verpackung",
```
Comment (ps-holler) [inline translations/de/wirecardpaymentgateway_de_lang.php]: We can use OXID's `WRAPPING` key instead.

File: translations/de/wirecardpaymentgateway_de_lang.php [hunk 4]
```diff
@@ -460,7 +462,9 @@
     "vault_use_existing_text" => "Gespeicherte Kreditkarte verwenden",
     "vault_use_new_text" => "Neue Kreditkarte verwenden",
     "vault_valid_until_text" => "Gültig bis",
+    "voucher" => "Gutschein",
```
Comment (ps-holler) [inline translations/de/wirecardpaymentgateway_de_lang.php]: We can use OXID's `COUPON` key instead.

File: translations/de/wirecardpaymentgateway_de_lang.php [hunk 5]
```diff
@@ -182,6 +182,7 @@
     "first-name" => "Vorname",
     "first_name_input" => "Vorname",
     "gender" => "Geschlecht",
+    "gift_card" => "Geschenkkarte",
```
Comment (ps-holler) [inline translations/de/wirecardpaymentgateway_de_lang.php]: We can use OXID's `GREETING_CARD` key instead.

File: Core/BasketHelper.php (hunk 1 above)
Comment (ps-cgrach) [reply Core/BasketHelper.php]: 👍

File: Extend/Payment_Gateway.php (hunk 2 above)
Comment (ps-cgrach) [reply Extend/Payment_Gateway.php]: 👍

**Pull request merged**